### PR TITLE
add batching and retries to kinesisExporter

### PR DESCRIPTION
**Pull request by sleepylemur**
@rmarianski mind giving this a look? also if you can think of any other tests i should do for the retry or batching functionality lemme know and i'll add em.

### CODE REVIEW

File: test/exporters/kinesisExporter.js [hunk 1]
```diff
@@ -21,20 +21,82 @@ function stubFirehose(fails) {
...
+        'aws-sdk': { Firehose: stubs.Firehose }
+      });
+      var exporter = new KinesisExporter({ region: 'oz', streamName: 'teststream' });
+
+      for (var i = 0; i < 499; i++) {
+        exporter.add('payload');
+      }
+      expect(stubs.putRecordBatch.called).to.be.false;
+      exporter.add('payload');
+      exporter.add('payload');
```
Comment (rmarianski) [inline test/exporters/kinesisExporter.js]: Do you have to add 2 here to get it to send?

File: test/exporters/kinesisExporter.js [hunk 2]
```diff
@@ -21,20 +21,82 @@ function stubFirehose(fails) {
...
+      expect(stubs.putRecordBatch.called).to.be.true;
+      expect(stubs.putRecordBatch.firstCall.args[0].Records.length).to.eq(500);
+    });
+
+    it('retries batch at expected intervals', function () {
+      var stubs = stubFirehose(true);
```
Comment (rmarianski) [inline test/exporters/kinesisExporter.js]: I guess `true` means to generate failures?

**Comment by rmarianski** [comment]
Just put in a couple of minor comments about the tests. Looks good to me :+1:

**Comment by sleepylemur** [comment]
@rmarianski thanks. the true = fail thing had been annoying me a little too so i switched it to true = works, false = fails... not great but maybe a little less confusing. also i took out the extra `add`, which i had in there intentionally to try to test that it wouldn't batch more than 500, but that was a silly way to test it.

**Pull request merged**